File: SparkCourses/20-02-2019/Section2/customer-orders.py
from pyspark import  SparkConf, SparkContext

conf = SparkConf().setMaster("local").setAppName("CustomerOrders")
sc = SparkContext(conf = conf)

def parseLine(lines):
    fields = lines.split(",")
    customer_id = fields[0]
    amount = fields[2]
    return (int(customer_id), float(amount))

lines = sc.textFile("/home/bjit-557/BigData/February/SparkCourses/20-02-2019/Section2/Assignment1/customer-orders.csv")
parsedLine = lines.map(parseLine)
results = parsedLine.reduceByKey(lambda x,y: x+y)
resultsSorted = results.map(lambda x:(x[1], x[0])).sortByKey().map(lambda x:(x[1], x[0]))
resultsSorted = resultsSorted.collect()
for result in resultsSorted:
    print(result)

File: scala/customer_orders.scala
import org.apache.spark.SparkContext
import org.apache.spark.SparkConf

object customer_orders{
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setMaster("local").setAppName("CustomerOrder")
    val sc = new SparkContext(conf)
    val srcFile = "../SparkCourses_Scala/data/customer-orders.csv"
    val lines = sc.textFile(srcFile)
    val parsedLines = lines.map(map_parseLine).reduceByKey(reduce_parseLine)
    val sortedResult = parsedLines.map(swap).sortByKey()
    sortedResult.foreach(Print)
  }

  def map_parseLine(x:String) : (Int, Float) = {
    val fields = x.split(",")
    val customer_id = fields(0).toInt
    val amount = fields(2).toFloat
    return (customer_id, amount)
  }

  def reduce_parseLine(x:Float, y:Float) : (Float) = {
    return x+y
  }

  def swap(x:(Int, Float)) : (Float, Int) = {
    return (x._2, x._1)
  }

  def Print(data:(Float, Int)) : Unit = {
    println("CustomerId : "+data._2)
    println("Amount : " +data._1)
    println("#########################")
  }
}
